### ISSUE TITLE
Tips writing lisp code in spacemacs?

### ISSUE THREAD

**Issue by alex-glv** [syl20bnr/spacemacs#1812]
This is not a bug report, merely request for some guidance.
Before spacemacs I used paredit for lisp and were quite efficient. I had never had issues with unmatched parens. Paredit was quite strict and to have unmatched parens, one had to C-q <char delete command>.
Even though I have smart parens strict enabled, I manage to get into unmatched parens.

One feature I relied upon a lot was "comment sexp". Now I have comment line / region (with SPC c l) but it's not as useful.

Navigating around sexps seems cumbersome, or did I just not get into habit yet?
e.g. I can go to the next closing paren, but not the previous. Is there an equivalent of "dive in" command?
Some commands I can't understand what they do:
 - Absorb expression
 - Convolute expression

What are your tips on editing lisp with evil? Do you jump between normal mode and evil lisp, or evil lisp exclusively? What are the must commands to be effective?

Thanks!

**Comment by syl20bnr** [comment]
evil-lisp-state is a WIP and I tried a lof of different UX (each major version of this package is a different UX). I tried to make lisp state that can replace the normal state but the result was never satisfying.
The current approach is to use lisp-state to manipulate sexp but not really to navigate, the `j` and `k` bindings are more a convenient way to go to a close parenthesis and then execute some operations.
But the current evil-lisp-state is sticking by default and a lot of users don't like it, I will experiment on an optional sticking feature.

I don't use absorb, convolute is hard to explain even-though I use it :-D It swaps the parent sexp with the grand-parent sexp.

If you are used to paredit then drop evil-lisp-state by adding it to `dotspacemacs-excluded-packages` and configure paredit to your liking.
